lib/Basics/FixedSizeAllocator.h

```diff
@@ -134,11 +144,21 @@ class FixedSizeAllocator {
   void allocateBlock() {
     // minimum block size is for 64 items
     // maximum block size is for 4096 items
-    size_t const numItems = 64 << std::min<size_t>(6, _blocks.size() + 1);
-    _blocks.emplace_back(std::make_unique<MemoryBlock>(numItems));
+    size_t const numItems = 64 << std::min<size_t>(6, _numBlocks + 1);
+
+    // assumption is that the size of a cache line is at least 64,
+    // so we are allocating 64 bytes in addition
+    auto const dataSize = sizeof(T) * numItems + 64;
+    void* p = ::operator new(sizeof(MemoryBlock) + dataSize);
+    new (p)MemoryBlock(numItems);
```

Comment:
If the constructor throws an exception here, we will leak the memory in `p`.
Currently the constructor does not throw, so it is safe.
But it would be good to either mark the constructor of MemoryBlock as `noexcept`, or at least leave a code comment here about it.